solutions/deploy-argocd-apps-pull/getting-started.md

```diff
@@ -0,0 +1,123 @@
+# Getting Started
+
+### Prerequisites
+- [kind](https://kind.sigs.k8s.io) must be installed on your local machine. The Kubernetes version must be >= 1.19, see [kind user guide](https://kind.sigs.k8s.io/docs/user/quick-start/#creating-a-cluster) for more details.
+- 
+
+### Steps
+1. Setup an OCM Hub cluster and registered an OCM Managed cluster. 
+   ```
+   curl -L https://raw.githubusercontent.com/open-cluster-management-io/OCM/main/solutions/setup-dev-environment/local-up.sh | bash
+   ```
+   
+   See [Open Cluster Management Quick Start](https://open-cluster-management.io/getting-started/quick-start/) for more details.
+
+2. Install ArgoCD on the hub cluster and both managed clusters. 
+   ```
+   kubectl config use-context kind-hub #repeat for kind-cluster1 and kind-cluster2
+   kubectl create namespace argocd
+   kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+   ```
+   See [ArgoCD website](https://argo-cd.readthedocs.io/en/stable/getting_started/) for more details.
+
+3. On the Hub cluster, scale down the Application controller:
+    ```
+    kubectl -n argocd scale statefulset/argocd-application-controller --replicas 0
+    ```
+    **Note:** This step is not necssary if the ArgoCD instance you are using contains the feature: 
+    https://argo-cd.readthedocs.io/en/latest/user-guide/skip_reconcile/
+
+1. Install the Pull controller:
+    ```
+    kubectl apply -f https://raw.githubusercontent.com/open-cluster-management-io/argocd-pull-integration/main/deploy/install.yaml
+    ```
+
+1. If your controller starts successfully, you should see:
+    ```
+    $ kubectl -n open-cluster-management get deploy | grep pull
+    argocd-pull-integration-controller-manager   1/1     1            1           106s
+    ```
+
+1. On the Hub cluster, create an ArgoCD cluster secret that represents the managed cluster. This step can be automated with [OCM auto import controller](https://github.com/open-cluster-management-io/multicloud-integrations/).
+
+    **Note**: replace the `cluster-name` with the registered managed cluster name.
+    ```
+    cat <<EOF | kubectl apply -f -
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: <cluster-name>-secret # cluster1-secret
+      namespace: argocd
+      labels:
+        argocd.argoproj.io/secret-type: cluster
+    type: Opaque
+    stringData:
+      name: <cluster-name> # cluster1
+      server: https://<cluster-name>-control-plane:6443 # https://cluster1-control-plane:6443
+    EOF
+    ```
+
+1. On the Hub cluster, apply the manifests in `example/hub`:
+    ```
+    kubectl apply -f example/hub
+    ```
+
+1. On the managed clusters, apply the manifests in `example/managed`:
+    ```
+    kubectl config use-context kind-cluster1 #repeat for kind-cluster2
```

Comment:
You can write a for loop like the above comment and hardcode the cluster1 and cluster2.